CHANGES.md

```diff
@@ -36,6 +36,16 @@ OpenSSL 3.5
 
   *Tomáš Mráz*
 
+* Added support for a new callback registration SSL_CTX_set_new_pending_conn_cb,
+  which allows for application notification of new connection SSL object
+  creation, which occurs independently of calls to SSL_accept_connection().
+  Note: Because QUIC holds a lock for the QUIC engine associated with a
+  connection, recursive calls into any SSL api from a callback function
+  should be considered unsafe, potentially resulting in deadlock, and should
```

Comment:
Perhaps `any` SSL API is too strong a constraint. It should be safe to add `ex_data` to the object for example, which is I think the expected use of the callback. That's an SSL API. Perhaps worth calling out explicitly?

Reply:
This is true, but, based on conversation around the need for this, its difficult to bifurcate calls that are safe vs unsafe in any reasonable manner.  you're example is a good one, but I thought it better to be overly conservative since we plan to address it in a future release anyway

Reply:
We must not be over conservative to the point that the callback loses its ability to be useful.

Reply:
This is getting a bit frustrating.  Without a complete enumeration of what API calls are considered safe vs unsafe, I can think of no language that creates a clear distinction between the two, we're going to run afoul of people who think something is safe that isn't or complain about not being able to use a call that actually is safe.  I've modified the CHANGES entry to indicate that calls that trigger I/O or mutate I/O related state should be avoided, which is the closest I can think to get.  If someone has better concise language to place here, please let me know.